python/vespa/vespa/package.py

```diff
@@ -468,3 +499,264 @@ def deploy(self, disk_folder: str, container_memory: str = "4G"):
+            last = update.get("lastId", last)
+
+            if update["active"]:
+                sleep(1)
+            else:
+                status = update["status"]
+                if status == "success":
+                    return
+                elif status == "error":
+                    raise RuntimeError(
+                        "Unexpected error during deployment; see log for details"
+                    )
+                elif status == "aborted":
+                    raise RuntimeError(
+                        "Deployment was aborted, probably by a newer deployment"
+                    )
+                elif status == "outOfCapacity":
+                    raise RuntimeError(
+                        "No capacity left in zone; please contact the Vespa team"
+                    )
+                elif status == "deploymentFailed":
+                    raise RuntimeError("Deployment failed; see log for details")
+                elif status == "installationFailed":
+                    raise RuntimeError("Installation failed; see Vespa log for details")
+                elif status == "running":
+                    raise RuntimeError("Deployment not completed")
+                elif status == "endpointCertificateTimeout":
+                    raise RuntimeError(
+                        "Endpoint certificate not ready in time; please contact Vespa team"
+                    )
+                elif status == "testFailure":
+                    raise RuntimeError(
+                        "Unexpected status; tests are not run for manual deployments"
+                    )
+                else:
+                    raise RuntimeError("Unexpected status '" + status + "'")
+
+    @staticmethod
+    def print_log_entry(step: str, entry: dict):
+        timestamp = strftime("%H:%M:%S", gmtime(entry["at"] / 1e3))
+        message = entry["message"].replace("\n", "\n" + " " * 23)
+        if step != "copyVespaLogs" or entry["type"] == "error":
+            print("{:<7} [{}]  {}".format(entry["type"].upper(), timestamp, message))
+
+    def deploy(self, instance: str, application_package: ApplicationPackage) -> Vespa:
+        """
+        Deploy the given application package as the given instance in the Vespa Cloud dev environment.
+
+        :param instance: Name of this instance of the application, in the Vespa Cloud.
+        :param application_package: ApplicationPackage to be deployed.
+
+        :return: a Vespa connection instance.
+        """
+        region = self.get_dev_region()
+        job = "dev-" + region
+        run = self.start_deployment(instance, job, application_package)
+        self.follow_deployment(instance, job, run)
+        endpoint_url = self.get_endpoint(
+            instance=instance,
+            region=region,
+            application_package_name=application_package.name,
+        )
+        return Vespa(url=endpoint_url, cert=self.data_cert_file.name)
```

Comment:
@jonmv It would be nice to include the `deployment_message` argument when creating the `Vespa` instance. What would best represent the deployment message in the Vespa Cloud case?

Reply:
I guess that depends. What is it used for? The whole log could be it. It's usually not that long for dev deployments. So, everything that's printed in `follow_deployment`.

Reply:
Ok, for now it is just for informational purposes.

Reply:
The deployment message from the config server is in there, if you do substring search, etc..